### PR TITLE
Header encoding fix

### PR DESCRIPTION
**Pull request by ffcactus**
Motivation:

Line folding should consider the Unicode that occupies more than one byte.

Conformance:

Your commits should be signed and you should have signed the Eclipse Contributor Agreement as explained in https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md.
Please also make sure you adhere to the code style guidelines: https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines

### CODE REVIEW

File: src/test/java/io/vertx/ext/mail/mailencoder/MailEncoderTest.java [hunk 1]
```diff
@@ -206,10 +206,10 @@ public void testToManyEncoded() throws Exception {
     String mime = new MailEncoder(message, HOSTNAME).encode();
     assertThat(mime, containsString("To: user0@example.com (=?UTF-8?Q?=C3=84a?=),user1@example.com\n"
       + " (=?UTF-8?Q?=C3=84a?=),user2@example.com (=?UTF-8?Q?=C3=84a?=),\n"
-      + " user3@example.com (=?UTF-8?Q?=C3=84a?=),user4@example.com (=?UTF-8?Q?=C3?=\n"
-      + " =?UTF-8?Q?=84a?=),user5@example.com (=?UTF-8?Q?=C3=84a?=),\n"
-      + " user6@example.com (=?UTF-8?Q?=C3=84a?=),user7@example.com (=?UTF-8?Q?=C3?=\n"
-      + " =?UTF-8?Q?=84a?=),user8@example.com (=?UTF-8?Q?=C3=84a?=),\n"
+      + " user3@example.com (=?UTF-8?Q?=C3=84a?=),user4@example.com (=?UTF-8?Q??=\n"
```
Comment (ffcactus) [inline src/test/java/io/vertx/ext/mail/mailencoder/MailEncoderTest.java:209]: As you can see, `=C3=84` should not fold into the next line.
But also note that we may have "**??=**\n" at the end of the line, but I think is allowed.

**Review by gaol — CHANGES_REQUESTED** [review]
Thanks @ffcactus for the contribution, I have several comments, please take a look.

**Comment by gaol** [comment]
Please make sure you are using correct email (good format and the one you are using to sign ECA) in the git commit, I see your email in the git commits is: `bin.bai@bin.bai@rakuten.com`, which looks not a valid email address..

And it would be better to squash your commits into 1. thanks :)

**Comment by ffcactus** [comment]
> Please make sure you are using correct email (good format and the one you are using to sign ECA) in the git commit, I see your email in the git commits is: `bin.bai@bin.bai@rakuten.com`, which looks not a valid email address..
> 
> And it would be better to squash your commits into 1. thanks :)

Should I squash the commits into a single one and force push or make a squashed PR that squashes the commits after the review?

**Comment by gaol** [comment]
@ffcactus you can squash and force push to update this PR.

**Comment by ffcactus** [comment]
Sorry, I think I messed up this PR. 
I'm going to discord this PR. Please refer to https://github.com/vert-x3/vertx-mail-client/pull/217

**Pull request closed without merging**